srcutil/gen_command_info.py

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env python
+import json
+import argparse
+import os.path
+
+
+class Scope:
+    def __init__(self, start, end, file):
+        self.file = file
+        self.start = start
+        self.end = end
+
+
+    def __enter__(self):
+        self.write(self.start + '{\n')
+        Scope.indent += 1
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        Scope.indent -= 1
+        self.write(f'}}{self.end}\n')
+
+    def write(self, line):
+        indent = '  ' * Scope.indent
+        self.file.write(indent + line)
+
+Scope.indent = 0
+
+def get_function_signature(name):
+    tokens = [token.title() for token in name.replace('.', ' ').split(' ')]
+    value = ''.join(tokens)
+    return f'int Set{value}Info(RedisModuleCtx *, RedisModuleCommand *cmd)'
+
+def generate_header_file(file_name, command_names):
+    with open(file_name + '.h', 'w') as header:
+        header.write('// This file is generated by gen_command_info.py\n')
+        header.write('#pragma once\n')
+        header.write('#include "redismodule.h"\n\n')
+        for name in command_names:
+            header.write(f'{get_function_signature(name)};\n')
+
+
+exclude_list = ['group']
+
+def generate_history(file, changes):
+    with Scope('.history = (RedisModuleCommandHistoryEntry[])', ',', file) as history:
+        for change in changes:
+            version, what = change
+            history.write(f'{{"{version}", "{what}"}},\n')
+        history.write('{0}\n')
+
+def generate_arguments(file, member, arguments):
+    min_arity = 0
+    with Scope(f'.{member} = (RedisModuleCommandArg[])', ',', file) as args_scope:
+        for arg in arguments:
+            with Scope('', ',', file) as arg_scope:
+                for string_arg in ['name', 'token', 'summary', 'since', 'deprecated_since', 'display_text']:
+                    if string_arg in arg:
+                        arg_scope.write(f'.{string_arg} = "{arg[string_arg]}",\n')
+                if 'type' in arg:
+                    arg_scope.write(f'.type = REDISMODULE_ARG_TYPE_{arg['type'].replace('-', '_').upper()},\n')
+                flags = []
+                if 'optional' not in arg:
+                    min_arity += 1
+                for flag in ['optional', 'multiple', 'multiple-token']:
+                    if flag in arg:
+                        flags.append(f'REDISMODULE_CMD_ARG_{flag.replace('-', '_').upper()}')
+                if len(flags) > 0:
+                    arg_scope.write(f'.flags = {' | '.join(flags)},\n')
+                if 'arguments' in arg:
+                    generate_arguments(file, 'subargs', arg['arguments'])
+        args_scope.write('{0}\n')
+    return min_arity
+
+
+def generate_redis_module_command_info(cmd_info, file):
+    with Scope('const RedisModuleCommandInfo info = ', ';', file) as info:
+        info.write('.version = REDISMODULE_COMMAND_INFO_VERSION,\n')
```

Comment:
Not the right version. Let's remove or have the correct one (I guess the first).

Reply:
See docs:
https://redis.io/docs/latest/develop/reference/modules/modules-api-ref/
![image](https://github.com/user-attachments/assets/5a681492-9fe4-4710-9e55-31e6dfe6548a)
It should always be set to `REDISMODULE_COMMAND_INFO_VERSION`